sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala

```diff
@@ -62,18 +62,27 @@ private[sql] object H2Dialect extends JdbcDialect {
           assert(f.children().length == 1)
           val distinct = if (f.isDistinct) "DISTINCT " else ""
           Some(s"STDDEV_SAMP($distinct${f.children().head})")
-        case f: GeneralAggregateFunc if f.name() == "COVAR_POP" =>
+        case f: GeneralAggregateFunc if f.name() == "COVAR_POP" && !f.isDistinct =>
           assert(f.children().length == 2)
-          val distinct = if (f.isDistinct) "DISTINCT " else ""
```

Comment:
does it mean all distinct functions are not supported now?

Reply:
No. Just H2 not supports `COVAR_POP` with distinct.

Reply:
so this is a bug fix?

Reply:
Yes. remove no need code.

Reply:
Can we have a separate PR for this? I think previously the query fails if the push down distinct COVAR_POP to H2

Reply:
OK